disttask/framework/dispatcher/dispatcher.go

```diff
@@ -414,33 +413,26 @@ func (d *dispatcher) processNormalFlow(gTask *proto.Task) (err error) {
 	}
 
 	// Generate all available TiDB nodes for this global tasks.
-	serverNodes, err1 := GenerateSchedulerNodes(d.ctx)
+	serverNodes, err1 := handle.GetEligibleInstances(d.ctx, gTask)
 	if err1 != nil {
 		return err1
 	}
+	if len(serverNodes) == 0 {
+		return errors.New("no available TiDB node")
+	}
 	subTasks := make([]*proto.Subtask, 0, len(metas))
 	for i, meta := range metas {
-		instanceID, err := GetEligibleInstance(serverNodes, i)
-		if err != nil {
-			logutil.BgLogger().Warn("get a eligible instance failed", zap.Int64("gTask ID", gTask.ID), zap.Error(err))
-			return err
-		}
+		instanceID := GetInstanceForSubtask(serverNodes, i)
 		subTasks = append(subTasks, proto.NewSubtask(gTask.ID, gTask.Type, instanceID, meta))
 	}
 	return d.updateTask(gTask, gTask.State, subTasks, retrySQLTimes)
 }
 
-// GetEligibleInstance gets an eligible instance.
-func GetEligibleInstance(serverNodes []*infosync.ServerInfo, pos int) (string, error) {
-	if pos >= len(serverNodes) && pos < 0 {
-		errMsg := fmt.Sprintf("available TiDB nodes range is 0 to %d, but request position: %d", len(serverNodes)-1, pos)
-		return "", errors.New(errMsg)
-	}
-	if len(serverNodes) == 0 {
-		return "", errors.New("no available TiDB node")
-	}
-	pos = pos % len(serverNodes)
-	return serverNodes[pos].ID, nil
+// GetInstanceForSubtask gets an eligible instance.
+// we assign the subtask to the instance in a round-robin way.
+func GetInstanceForSubtask(serverNodes []*infosync.ServerInfo, taskIdx int) string {
+	pos := taskIdx % len(serverNodes)
+	return serverNodes[pos].ID
 }
 
 // GenerateSchedulerNodes generate a eligible TiDB nodes.
```

Comment:
Prefer move this function to TaskHandle, https://github.com/pingcap/tidb/blob/master/disttask/framework/dispatcher/dispatcher.go#L63, and pass the handle to bussiness dispatcher

Reply:
let's change it when we support "subtask not bind to instance"